File: java/program/Lizard.java
public class Lizard extends Reptile {
    private boolean tailRegrow;

    public Lizard(String name, double length, String habitat, String diet, boolean tailRegrow) {
        super(name, length, habitat, diet);
        this.tailRegrow = tailRegrow;
    }

    @Override
    public void display() {
        System.out.println("Lizard: " + name +
                           ", Tail can regrow: " + (tailRegrow ? "Yes" : "No") +
                           ", Habitat: " + habitat);
    }
}

File: cpp/Lizard.cpp
#include "Lizard.h"

Lizard::Lizard(string n, double l, string h, string d, bool t)
    : Reptile(n, l, h, d), tailRegrow(t) {}

void Lizard::display() {
    cout << "Lizard: " << name
         << ", Tail can regrow: " << (tailRegrow ? "Yes" : "No")
         << ", Habitat: " << habitat << endl;
}
